### ISSUE TITLE
Move pytest to Github

### ISSUE THREAD

**Issue by pytestbot** [pytest-dev/pytest#699]
Originally reported by: **Leonardo Santagada (BitBucket: [santagada](http://bitbucket.org/santagada), GitHub: [santagada](http://github.com/santagada))**

----------------------------------------

Why isn't pytest on github?

There is no explanation anywhere about why not use git/github so I thought it is a good idea to have this explanation here and maybe moved to a faq on the site or something.

I don't really want to force anyone to move, but I would guess that there more people like me that would like to know if there is any advantages for the project in being on bitbucket.

----------------------------------------
- Bitbucket: https://bitbucket.org/pytest-dev/pytest/issue/699

**Comment by bubenkoff** [comment]
@nicoddemus you did a great job!

**Comment by pytestbot** [comment]
Thanks! :smile:

**Comment by Themanwithoutaplan** [comment]
Just for the record, I've yet to see any compelling reasons for Github over BitBucket. Travis can easily be run using a hg-git as can pretty much every other service. Bitbucket's ecosystem, via Atlassian, is arguably richer. If people are incapable of contributing to a project because it's not on Github then their contributions are unlikely to be any good.

And, personally, I find Github's UI reminds me of nursery school. But these things are intensely personal.

Still, what's done is done.

**Comment by RonnyPfannschmidt** [comment]
in general terms bitbucket is much worse in terms of ui, contributor turnaround and sane tooling

we have no hard data, but its what we perceived in anecdotes over the last few months before and after the switch

we would generally prefer hg, but bitbucket currently is a a reason to leave behind the better tool for a better surrounding setup

github did perfectly put itself into a market position of being a opensource hub while bitbucket was going for "killing" hg instead

even i (who is the main proponent for staying on hg) can easily agree on github being better than bitbucket

**Comment by Themanwithoutaplan** [comment]
I didn't want to warm up an old discussion: what's done is done.

I'm pretty opinionated on UI but it's best to leave it to personal preferences. I don't understand the tooling arguments, maybe because we run our own Jenkins for CI.

I suspect what a lot of people don't know, however, is that Atlassian provides free JIRA instances for open source projects.

**Comment by hpk42** [comment]
FWIW my 2cents: i certainly don't see the github UI as superior to bitbucket.  I am rather missing some things like voting and agree with @Themanwithoutaplan that the github UI is unneccessarily flashy in sokme places.  I do appreciate automatic PR testing which we didn't have with BB (but maybe it could have been done).  Also it seems more people are familiar with github workings which is a bit of a plus although i generally am skeptical of contributing to furthering monopolies.

**Comment by warsaw** [comment]
On Jul 14, 2015, at 12:59 PM, hpk42 wrote:

>I do appreciate automatic PR testing which we didn't have with BB (but maybe
>it could have been done).  Also it seems more people are familiar with github
>workings which is a bit of a plus although i generally am skeptical of
>contributing to furthering monopolies.

Not that my opinion counts for anything, but we've moved Mailman 3 to gitlab,
and I've moved several of my personal projects there.  It's a nice git hosting
option, it's free software (the community edition is anyway), and does have
some CI integration (see the Mailman project for some details).  It may not be
as feature full as github currently, but it's completely usable and easy, so
we're happy with it.
